qemu/qmp.go

```diff
@@ -1177,6 +1178,12 @@ func (q *QMP) ExecuteCPUDeviceAdd(ctx context.Context, driver, cpuID, socketID,
 		args["thread-id"] = threadID
 	}
 
+	if q.version.Major > 4 || (q.version.Major == 4 && q.version.Minor >= 1) {
+		if dieID != "" {
```

Comment:
@devimc  I think you could simply check the driver type here. Something like `&& driver=="x86_64"` . Do you know if dieID is valid for power and arm too?

Reply:
Whatever we do, we probably want to have one mechanism for handling socket, thread and die.  Either we handle all three here or we handle them in virtcontainers.

Reply:
Ok, agree. If you want I can open a small PR to do that. Something like:
```golang
bool isSocketSupported(driver string)  {
        if driver == "host-s390x-cpu" && driver "host-ppc64le-cpu" {
                return false
        }
        return true
}
```
and then in 
```golang
        if isSocketSupported(driver) {
                        args["socket-id"] = socketID

        }

```
we could do that for all 3 param

Reply:
@devimc, @sboeuf What do you guys think?  Should we move the platform specific [checks](features2d_20190717-110936.xml) in virtcontainers into govmm?

Reply:
@markdryan sounds good, I will update this pr

Reply:
@devimc There won't actually  be a problem until virtcontainers is updated to pass the die-id parameter.  We could merge this commit as is, and then have a separate commit that adds the s390x checks for socket, thread and die to govmm, and finally, update the code in virtcontainers.

Reply:
@alicefr  @markdryan  got it, I have filed an issue https://github.com/intel/govmm/issues/102

Reply:
So any objection if I merge this?  @devimc Who will make the kata changes for die-id?  Will that be you?  If so, we should be good as long as we fix #102 before adding die support to virtcontainers.

Reply:
I think I need a confirmation for arm and power if the die-id is used for those archs. I guess not for power, but I don't know for arm.

Reply:
@markdryan yes I'm going to patch kata, we have time since QEMU 4.1 is not out yet :smiley:

Reply:
I can do the PR for socketid, threadID and die-id, if you can help me to understand the if the options are supported for the other archs

Reply:
@alicefr Great.  Thanks.

@devimc Any idea who would know about the other architectures?

Reply:
A quick search through the qemu code shows that it might just be x86

https://github.com/qemu/qemu/search?q=has_die_id&unscoped_q=has_die_id

Reply:
@markdryan @alicefr `die` is x86 specific, see https://github.com/qemu/qemu/commit/1b45842203540943b1e22fc30764456c035d8637 and https://github.com/qemu/qemu/commit/c26ae610811e8d52f4fc73e3ae0a8bc4a24d6763

Reply:
/cc @nitkon (power) and /cc @Pennyzct (arm). Could you help me with that? Also will be nice if you could give me the exect name of the cpu driver for the cpu-hotplug like `host-s390x-cpu`

Reply:
@alicefr I asked in the QEMU irc channel and they told me that currently it's only implemented in x86, but it doesn't mean it won't appear in other archs in the future

Reply:
ok then, in the future will be enough to update the function: For me the suppor table looks like

| Arch | cpu driver             | socker id | thread id | die id |
| ---     |  ---                         |      ---        |   ---         |  --       |
| x86 |    host-x86_64-cpu | y               |   y           | y        |
| s390x |        host-s390x-cpu | n | n |n |
| arm64  |                           |                   |                |            |
| ppc64le |                          |                   |                |           |

Reply:
Hi~ @alicefr @justin-he is working on this filed~ ;)

Reply:
> ok then, in the future will be enough to update the function. For me the support table looks like
> 
> Arch	cpu driver	socket id	thread id	die id
> x86_64	host-x86_64-cpu	y	y	y ( >= Qemu 4.1)
> s390x	host-s390x-cpu	n	n	n
> arm64		y	y	n
> ppc64le	host-powerpc64-cpu	n	n	n

Hi @alicefr
The cpu driver of arm64 in qemu (virt machine type) should be "driver=host-arm-cpu"
But it is worth to note that the cpu hotplug (what kata required) has not been supported yet.

Reply:
@justin-he thanks,  @alicefr  @markdryan can we merge this?

Reply:
@devimc for me yes, I'll open a PR for the threadID, socketID and dieID

Reply:
I open a PR https://github.com/intel/govmm/pull/103 with the changes with the check for the 3 params